### ISSUE TITLE
Request: Word count?

### ISSUE THREAD

**Issue by varlokkur** [jamesjonesmath/canvancement#22]
Sorry if this isn't the best way to communicate, but is it possible to create a word count tool for speedgrader?  This feature has been requested for some time (https://community.canvaslms.com/ideas/3454-word-count-for-speedgrader-docviewer-grading) but they don't show any signs of implementing it.

**Comment by br2490** [comment]
POC:
document.querySelector('.Pages').innerText.split(" ").length
or
var count = 0; Object.values(document.getElementsByClassName('Page-container')).map(x => count += x.innerText.split(" ").length); console.log(count);

**Comment by jamesjonesmath** [comment]
@varlokkur, what are you wanting a word count of? Text entry? PDF? Quizzes? Essay quiz questions? Total for a discussion? Individual posts for a discussion? There's a lot of stuff that goes into SpeedGrader and I don't think it's a simple solution.

For a text entry submission, the content is inside the element with id='submission_preview'. To build from @br2490 suggestion, 
```
var wordCount = document.getElementById('submission_preview').innerText.split(' ').length;
```

There's an comment that someone left in the idea you linked to that said they were able to do it for docx files, but that was for Crocodoc, not for DocViewer. When I pulled up a document in DocViewer, the .Pages or .Page-container class contains it. That's what Benjamin was referring to in his proof of concept. There's one of those for each page, though. So you would need document.querySelectorAll() instead of document.querySelector(). 

Quizzes and Discussions pull their information in the format of a quiz or discussion, so the CSS selector varies depending on the type of content.

So, yes, I think it would be possible, but it's not going to be trivial. You would need to decide where you wanted the output to go. console.log() is good for testing, but not for people actually using SpeedGrader. If you go with essay questions in quizzes or individual discussion posts, then you would need to allow for multiple occurrences within SpeedGrader.

You would also need to wait for the content to be loaded before you try to do a word count. QuizWiz might be a good starting point for someone looking how to do that. I did find a bug in QuizWiz tonight when I started researching this question. It seems that the speedgrader_iframe isn't always available when I thought it was.

The discussion might be better in the Community where others can contribute. It may turn out that someone else has already done it.
